code/globalincs/memory/memory.h

```diff
@@ -46,3 +46,24 @@ inline void *vm_realloc(void *ptr, size_t size)
 
 	return ret_ptr;
 }
+
+
+// For use with unique_ptr
+template <typename T>
+struct FreeDeleter
+{
+	void operator()(T* const p) const
+	{
+		free(p);
+	}
+};
+
+// For use with unique_ptr
+template <typename T>
+struct VmFreeDeleter
+{
+	void operator()(T* const p) const
+	{
+		vm_free(p);
+	}
+};
```

Comment:
I suggest actually adding something like this, since this is a semi-common use case:
```cpp
//Managed unique pointer for data allocated using vm_malloc. Never use with data allocated with new or malloc
template<typename T> using SCP_vm_unique_ptr = std::unique_ptr<T, VmFreeDeleter<T>>;
```